src/applications/Applications.ExtractImages/ExtractImagesHost.cs

```diff
@@ -90,7 +91,7 @@ public override void Start()
             var parser = new CohortCsvParser(_fileSystem);
             (ExtractionKey extractionKey, List<string> idList) = parser.Parse(_csvFilePath);
 
-            _extractionMessageSender.SendMessages(extractionKey, idList);
+            _extractionMessageSender.SendMessages(_absoluteExtractionDir, extractionKey, idList);
```

Comment:
Why do we pass the dir to `_extractionMessageSender`? Seems like a lot of code change in this PR is to allow the directory creation to happen in the `ExtractionMessageSender` class instead of  `ExtractImageHost`.  Whats the logic behind moving it?

Reply:
This solves #913 by moving the directory creation to after the confirmation prompt. The `SendMessages` function is currently the most obvious place to put that. The next PR #915 actually simplifies this further, so I'm not too worried about things looking perfect here.

Reply:
ok if it is  changing in the next PR anyway can go with it.